tm-monitor/docs/Design-proposal.md

```diff
@@ -0,0 +1,53 @@
+# Design proposal for Tendermint Monitor
+
+Tendermint monitor is designed to be the tool for all related parties to have a close look about the network. 
+
+## Design Goal for monitoring the network
+
+The design goal of Tendermint Monitor is to track **Liveness** & **Safety** of the whole network and certain validator.
+
+ The liveness of the network is in jeopardy if:
+
+* many validators are offline
+* the system barely gather +2/3 votes in recent blocks
+* validators have to wait for many rounds before commiting new block
+
+
+The safety of the network is in jeopardy if:
+
+* There exists a monopoly of Atom token, +1/3
+
+* Not enough Atoms are bonded
+
+* Certain validator attracts too many delegation
+
+
+## Design Goal for monitoring a validator
+
+The validator must monitor its **availability** and **safety** closely.  There are some early-stage indicators to be noticed:
+
+*  Are most of previous precommits included in blocks?
+* How many proposals it missed before?
+
+
+The safety of a validator is under pressure if :
+
+* it's not connected with enough persistent peers 
+* there is not enought peers in its address book
+
+## Tendermint Monitor Atchitecture
+
+The architecture of new version of monitor will be like this:
+
+![WechatIMG114](https://github.com/bianjieai/tendermint-tools/raw/bianjie/tm-monitor/tm-monitor/docs/pics/WechatIMG114.jpeg)
+
+The data will be stored in Mongo to facilitate better querying.
+
+## Metrics 
+
+![metrics](https://github.com/bianjieai/tendermint-tools/raw/bianjie/tm-monitor/tm-monitor/docs/pics/metrics.png)
+
+
+
+link: https://docs.google.com/spreadsheets/d/13TbMrG0PlahN-vr_Z8hbQJIQHxqM0g4C4AfF-kDYHPI/edit?usp=sharing 
```

Comment:
Great table! The first time I actually see a requirements document for monitoring. I think we can implement a decent percentage of this without ever creating any code, just reusing the Elastic, the TICK stack or Prometheus.
Note, that validators might want to create their own monitoring solutions and we're not supposed to give them "complete and supported" solutions for this. Only suggestions and examples. This is why I think creating multiple, reusable examples is better than one monolithic piece of code.